app-modules/portal/routes/api.php

```diff
@@ -52,19 +54,23 @@
     ->name('api.')
     ->middleware([
         'api',
+        NeedsTenant::class,
         EnsureKnowledgeManagementPortalIsEnabled::class,
         EnsureKnowledgeManagementPortalIsEmbeddableAndAuthorized::class,
     ])
     ->group(function () {
-        Route::middleware(['auth:sanctum'])->group(function () {
-            Route::get('/user', function (Request $request) {
-                $user = auth('student')->user() ?? auth('prospect')->user() ?? $request->user();
-
-                if (! auth('web')->check()) {
-                    return $user;
-                }
-            })->name('user.auth-check');
-        });
+        Route::middleware(['auth:sanctum'])
+            ->group(function () {
+                Route::get('/user', function (Request $request) {
+                    $user = $request->user('student') ?? $request->user('prospect') ?? null;
+
+                    if ($user?->tokenCan('knowledge-management-portal')) {
+                        return $user;
+                    }
+
+                    abort(Response::HTTP_FORBIDDEN);
```

Comment:
Wouldn't applying the `abilities:knowledge-management-portal` middleware to this route do this logic for you?